server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java

```diff
@@ -98,6 +99,13 @@ public SignificantTextAggregatorFactory(String name,
         this.significanceHeuristic = significanceHeuristic;
     }
 
+    private static boolean supportsAgg(MappedFieldType ft) {
+        if (ft.getTextSearchInfo() == TextSearchInfo.NONE) {
+            return false;
+        }
+        return ft.getTextSearchInfo() != TextSearchInfo.SIMPLE_MATCH_WITHOUT_TERMS;
```

Comment:
should it be return ft.getTextSearchInfo != NONE && gt.getTextSearchIngo != SIMPLE_MATCH_WITHOUT_TERMS ?

Reply:
++